### ISSUE TITLE
Captain Blackheart's Treasure's "discover" effect sometimes gives you duplicate card choices

### ISSUE THREAD

**Issue by Xinhuan** [HearthSim/hs-bugs#267]
https://twitter.com/ywoo_dev/status/667091159299174401

Confirmed to be a bug by Yong Woo.

**Comment by jleclanche** [comment]
Yeah I saw that too. So does this mean the brawl's discover is not a real discover?

**Comment by PattuX** [comment]
4.1: http://imgur.com/JV2iFva

**Comment by Xinhuan** [comment]
After watching a lot of screenshots and videos, I've come to the conclusion that the three choices offered in this Brawl are always in the order "Spell - Minion - Minion". Weapons are not offered in this Brawl. Since it is not a standard Discover effect, It is possible for the 2 offered minions to be duplicate minions.

Here is a link to Kripp's video of the Tavern Brawl, where in the first match, he is extremely surprised that the opponent Shaman managed to Discover something like 10 spells in a row:  https://www.youtube.com/watch?v=XXvl1FTmkxM

If your hero is replaced with Ragnaros, the first two choices are minions. It is unclear what the third choice (on the right) can be, but the non-collectible 2/3 vanilla Pirate can be offered in that slot, along with an Excavated Evil that was shuffled earlier into the player's deck: https://www.reddit.com/r/hearthstone/comments/423g88/discover_brawl_and_ragnaros_appear_to_have_an/

This leads to the speculation that the third slot might be a card that is remaining in your deck.

**Comment by PattuX** [comment]
If you're in fatigue as Ragnaros, you will only get offered 2 cards (with proof): 

https://www.reddit.com/r/hearthstone/comments/42b2os/so_this_is_what_happens_when_you_try_to_discover/

Sidenote: Jaraxxus always offers you Warlock cards.


Also the spell has the highest id among the 3 choices, so I guess that's an info-leak. You may decide whether it's worth opening another issue.

**Comment by Xinhuan** [comment]
The OP directly contradicted you: "If you don't pick the pirate (or the card drawn from your deck) your deck does not deplete."

**Comment by PattuX** [comment]
Unfortunatelly, I didn't take any proof, but @culinko can confirm.

**Comment by culinko** [comment]
The card is already subtracted from your deck when being presented with the 3 choices at the start of the turn. Does not matter if you pick it or not, it's already gone from the deck. We specifically checked that. No screenshots, though.

**Comment by Xinhuan** [comment]
https://www.youtube.com/watch?v=ZLBMNjBmXRc

The game respects the 60 card deck size limit when trying to insert these 3 cards at the start of the turn, so if you have a 60, 59 or 58 card deck, some or all of these card insertions fail and result in 3, 2 or 1 card being offered to you respectively from the top of your deck. In this video, the only minion in the deck is the 2/3 vanilla Pirate available only in the Tavern Brawl.

**Comment by Patashu** [comment]
This brawl came back, and a user is reporting that they hanged at the start of their turn as Ragnaros:

https://www.reddit.com/r/hearthstone/comments/4zdfw6/becoming_ragnaros_in_the_new_tavern_brawl_is/
